### ISSUE TITLE
Linux system compatibility 

### ISSUE THREAD

**Issue by Abdelnaby-M** [haf-decent/arpping#6]
Hello there, I just start using few weeks ago with windows platform and it was perfect, However when trying to implement in any linux based system it returns that error. could you help me figure it out.  

function Arpping({ timeout = 5, includeEndpoints = false, useCache = true, cacheTimeout = 3600 }) {
                   ^

TypeError: Cannot destructure property `timeout` of 'undefined' or 'null'.
    at new Arpping (/home/pinerium/node_modules/arpping/index.js:29:17)
    at Object.<anonymous> (/home/pinerium/apr.js:2:15)

**Comment by haf-decent** [comment]
I tried to get a little fancy with some destructuring assignment a week or two ago and forgot to provide a fallback object in the case that you try doing `new Arpping()` (no arguments) as opposed to `new Arpping({})` (empty object argument). 

Just updated, tested, and pushed, so it should be good to go now. Pull the changes and let me know if that fixes the issue.

**Comment by Abdelnaby-M** [comment]
const Arpping = require('arpping')({});

getArpTable(ip = null) {
      return new Promise((resolve, reject) => {
   	 Arpping.discover(ip, (err, hosts) => {
   	   if (err) {
   		 return reject(err);
   	   }
	   return (hosts);
   	 });
      });
    }

const table = await this.getArpTable(ip);
		this.log("table: ", table);



/node_modules/arpping/index.js:31
    this.timeout = parseInt(timeout) || timeout.toFixed(0);
                 ^

TypeError: Cannot set property 'timeout' of undefined
    at Arpping (/node_modules/arpping/index.js:31:18)

**Comment by haf-decent** [comment]
Which package version are you on? It looks like you're trying to use the old version syntax (where you initialize the function straight away `const arpping = require('arpping')()`), but there has been an update since you last installed the package that moved to a new initialization procedure:

```javascript
const Arpping = require('arpping');
const arpping = new Arpping({ timeout: 2 });
// you can also do `new Arpping({})` or, in the newest version I just pushed to npm, `new Arpping()`
```

See if that helps. I would also suggest running an `npm update` as I just published a new minor package version that lets you initialize the `Arpping` object without the need to pass it an object.

**Comment by Abdelnaby-M** [comment]
I did edit the initialization but still i am getting "800a03ea javascript syntax error".

const Arpping = require('arpping');
const arpping = new Arpping();
(async () => {
	const ip =  '10.2.100.113';
arpping.discover(ip, (err, hosts) => {
    if (err) return console.log(err);
    console.log(hosts); 
});
})();

**Comment by haf-decent** [comment]
Did you update to the newest version (1.1.0)?

**Comment by Abdelnaby-M** [comment]
yes i did

**Comment by haf-decent** [comment]
And what's the exact error you're getting now? The previous error was definitely due to the way you were initializing the object, but the latest snippet you sent looks like it should be working.

Edit: If it's the same `cannot set property 'timeout' of undefined`, then it looks like the `this` designation might not be assigned properly. Would you be able to share a little bit more of the surrounding code? From what I can see in previous snippets, it looks like you might be using arpping within another class/function, but I am unable to reproduce the error no matter how I wrap it.

**Comment by Abdelnaby-M** [comment]
I am working on an Homey-app that uses ewelink api to control sonoff devices locally 
below is the code from my driver 


const Arpping = require('arpping');
var arpping = new Arpping({});

	getArpTable(ip = null) {
      return new Promise((resolve, reject) => {
   	 arpping.discover(ip, (err, hosts) => {
   	   if (err) {
   		 return reject(err);
   	   }
	   return (hosts);
   	 });
      });
    }


async onInit() {
		this.log('MyDriver has been inited');
		const ip = await ManagerCloud.getLocalAddress();
		 this.log("ip: ", ip);
		const table = await this.getArpTable(ip);
		this.log("table: ", table);
}

using this code it actually doesn't return anything, not even an error 
trying it on my Windows It returns an empty object and java script syntax error
error code: 800A03EA
i tried fixing it with various methods but with no luck

**Comment by haf-decent** [comment]
In your getArpTable function, you need to `return resolve(hosts);` or else you won't actually get a result

**Comment by Abdelnaby-M** [comment]
actually it returns an empty list now and getting the same error when trying it on windows

**Comment by haf-decent** [comment]
Is there a reason you're not defining your functions using the 'function' keyword? Like `function getArpTable(ip = null) {}` and `async function onInit() {}`. If I try running the code you've sent, it throws an error for those functions. Not sure why it would return anything if that's the problem though.

**Comment by Abdelnaby-M** [comment]
yeah, In homey app i don't have to defined function as "function function_name( ) { }" 
here's another code i am running from the cmd 

const Arpping = require('arpping');
const arpping = new Arpping();

const ip =  '10.2.100.113';

async function getArpTable(ip = null){
      return new Promise(async (resolve, reject) => {
   	 arpping.discover(ip, (err, hosts) => {
   	   if (err) {
   		 return reject(err);
   	   }
	   return resolve(hosts);
   	 });
      });
    }


 getArpTable(ip);

**Comment by haf-decent** [comment]
Ok, and you're sure that it's this package that is causing the problem? I'm not sure what else to suggest other than just double-checking that the package is actually installed locally. 

Also, not that I think it will help, but I just published a new version (v2.0.0) that uses Promises so you no longer have to wrap the methods manually.

**Comment by haf-decent** [comment]
Hey @Abdelnaby-M,

Just published a new release today (#12). It kind of restructures the whole build, and as I haven't heard back on this issue, I'm going to close it. Feel free to check out the new release and let me know if you run into this or any other problem.
